fvm/src/syscalls/linker.rs

```diff
@@ -4,30 +4,20 @@ use std::mem;
 
 use fvm_shared::error::ErrorNumber;
 use fvm_shared::sys::SyscallSafe;
-use wasmtime::{Caller, Linker, WasmTy};
+use wasmtime::{Caller, WasmTy};
 
 use super::context::Memory;
 use super::error::Abort;
 use super::{charge_for_exec, update_gas_available, Context, InvocationData};
 use crate::call_manager::backtrace;
 use crate::kernel::{self, ExecutionError, Kernel, SyscallError};
 
-/// Binds syscalls to a linker, converting the returned error according to the syscall convention:
-///
-/// 1. If the error is a syscall error, it's returned as the first return value.
-/// 2. If the error is a fatal error, a Trap is returned.
-pub trait BindSyscall<Args, Ret, Func> {
-    /// Bind a syscall to the linker.
-    ///
-    /// 1. The return type will be automatically adjusted to return `Result<u32, Trap>` where
-    /// `u32` is the error code.
-    /// 2. If the return type is non-empty (i.e., not `()`), an out-pointer will be prepended to the
-    /// arguments for the return-value.
-    ///
-    /// By example:
-    ///
-    /// - `fn(u32) -> kernel::Result<()>` will become `fn(u32) -> Result<u32, Trap>`.
-    /// - `fn(u32) -> kernel::Result<i64>` will become `fn(u32, u32) -> Result<u32, Trap>`.
+/// A "linker" for exposing syscalls to wasm modules.
+#[repr(transparent)]
```

Comment:
I assume the `#[repr(transparent)]` is required for the `Linker` struct here as that makes the compiler guarantee that the `wasmtime::Linker<InvocationData<K>>`  is laid out the same as this `Linker<K>`?

Reply:
It is. Although it's not strictly necessary now that we're not casting. I'll remove it (we can always add it back).